test/k8sT/Services.go

```diff
@@ -115,27 +116,35 @@ var _ = Describe("K8sServicesTest", func() {
 	Context("Checks ClusterIP Connectivity", func() {
 
 		var (
-			demoYAML = helpers.ManifestGet("demo.yaml")
+			demoYAML  = helpers.ManifestGet("demo.yaml")
+			clusterIP string
+			appPods   map[string]string
 		)
 
-		BeforeEach(func() {
+		BeforeAll(func() {
 			res := kubectl.Apply(demoYAML)
-			res.ExpectSuccess("unable to apply %s: %s", demoYAML, res.CombineOutput())
+			res.ExpectSuccess("unable to apply %s", demoYAML)
+
+			err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			Expect(err).Should(BeNil(), "Pods are not ready after timeout")
+
+			clusterIP, _, err = kubectl.GetServiceHostPort(helpers.DefaultNamespace, serviceName)
+			Expect(err).Should(BeNil(), "Cannot get service %s", serviceName)
+			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
```

Comment:
Put the value of `clusterIP` in the string as users will only see 'expected <false> to be <true>' without seeing what the actual value was.